desc/geometry/surface.py

```diff
@@ -1146,3 +1146,114 @@ def get_axis(self):
         data = self.compute(["R", "Z"], grid=grid)
         axis = FourierRZCurve(R_n=data["R"][0], Z_n=data["Z"][0], sym=self.sym)
         return axis
+
+
+def _constant_offset_surface(
+    base_surface, offset, grid, M, N, full_output=False, params=None
+):
+    """Create a FourierRZSurface with constant offset from the base surface (self).
+
+    Implementation of algorithm described in Appendix B of
+    "An improved current potential method for fast computation of
+    stellarator coil shapes", Landreman (2017)
+    https://iopscience.iop.org/article/10.1088/1741-4326/aa57d4
+
+    NOTE: Must have the toroidal angle as the cylindrical toroidal angle
+    in order for this algorithm to work properly
+
+    NOTE: this function lacks the checks of the constant_offset_surface
+    so that it is jittable/differentiable
+
+    Parameters
+    ----------
+    base_surface : FourierRZToroidalSurface
+        Surface from which the constant offset surface will be found.
+    offset : float
+        constant offset (in m) of the desired surface from the input surface
+        offset will be in the normal direction to the surface.
+    grid : Grid, optional
+        Grid object of the points on the given surface to evaluate the
+        offset points at, from which the offset surface will be created by fitting
+        offset points with the basis defined by the given M and N.
+        If None, defaults to a LinearGrid with M and N and NFP equal to the
+        base_surface.M and base_surface.N and base_surface.NFP
+    M : int, optional
+        Poloidal resolution of the basis used to fit the offset points
+        to create the resulting constant offset surface, by default equal
+        to base_surface.M
+    N : int, optional
+        Toroidal resolution of the basis used to fit the offset points
+        to create the resulting constant offset surface, by default equal
+        to base_surface.N
+    full_output : bool, optional
+        If True, also return a dict of useful data about the surfaces and a
+        tuple where the first element is the residual from
+        the root finding and the second is the number of iterations.
+
+    Returns
+    -------
+    offset_surface : FourierRZToroidalSurface
```

Comment:
todo: update this (does not return surface, only the points that would then be fit)